### PR TITLE
Database connect and statement timeouts

### PR DESCRIPTION
**Pull request by spennihana**

### CODE REVIEW

**Review by jingsong — APPROVED** [review]
shipit. i need moar sleep.

**Review by jingsong — APPROVED** [review]
beboop beboop

File: airflow/settings.py [hunk 1]
```diff
@@ -183,6 +183,22 @@ def configure_orm(disable_connection_pool=False):
...
     engine_args['encoding'] = engine_args['encoding'].__str__()
 
+    # Sane timeouts for connections and statements
+    connect_args = {}
+
+    try:
+        connect_timeout_seconds = conf.getint('core', 'SQL_ALCHEMY_CONNECT_TIMEOUT_SECONDS')
+    except conf.AirflowConfigException:
+        connect_timeout_seconds = 30
+    connect_args['connect_timeout'] = connect_timeout_seconds
```
Comment (samschlegel) [inline airflow/settings.py]: Do we know if this works for all databases that are currently supported (SQLite, MySQL, Postgres)

File: airflow/settings.py [hunk 2]
```diff
@@ -183,6 +183,22 @@ def configure_orm(disable_connection_pool=False):
...
+
+    try:
+        statement_timeout_seconds = conf.getint('core', 'SQL_ALCHEMY_STATEMENT_TIMEOUT_SECONDS')
+    except conf.AirflowConfigException:
+        statement_timeout_seconds = 30
+    connect_args['options'] = '-c statement_timeout={}'.format(statement_timeout_seconds * 1000)
```
Comment (samschlegel) [inline airflow/settings.py]: This only works when Postgres is the backend. MySQL has a different thing iirc, and SQLite doesn't really have connections. Can just gate this to Postgres only for now probably, with a note that support should be added for MySQL

**Review by samschlegel — APPROVED** [review]

**Review by samschlegel — CHANGES_REQUESTED** [review]
oops meant to request changes

File: airflow/config_templates/default_airflow.cfg [hunk 3]
```diff
@@ -102,6 +102,18 @@ sql_alchemy_pool_size = 5
...
 sql_alchemy_pool_recycle = 1800
 
+# How many seconds to wait before initial connect times out
+sql_alchemy_connect_timeout_seconds = 30
+
+# How many seconds to wait before statements time out
+sql_alchemy_statement_timeout_seconds = 30
+
+# Maximum number of attempts to perform statement on the database
+sql_alchemy_statement_max_retries = 10
```
Comment (samschlegel) [inline airflow/config_templates/default_airflow.cfg]: These configs should be moved to the retrying PR

File: airflow/config_templates/default_airflow.cfg [hunk 4]
```diff
@@ -102,6 +102,18 @@ sql_alchemy_pool_size = 5
 # a lower config value will allow the system to recover faster.
 sql_alchemy_pool_recycle = 1800
 
+# How many seconds to wait before initial connect times out
+sql_alchemy_connect_timeout_seconds = 30
+
+# How many seconds to wait before statements time out
+sql_alchemy_statement_timeout_seconds = 30
```
Comment (samschlegel) [inline airflow/config_templates/default_airflow.cfg]: Both of these timeouts should default to 0 which would disable. Enabling this by default could be very surprising and potentially break things

**Comment by samschlegel** [comment]
Do you know if setting them to 0 will actually disable on the SQLAlchemy side? Or do we need to just remove the flag when it's 0

**Pull request closed without merging**